### PR TITLE
#21 Use synchronizedList to capture log events

### PR DESCRIPTION
**Pull request by AndreyNudko**
This makes LogCaptor suitable for using in multi-threaded 'integration' tests

### CODE REVIEW

File: src/main/java/nl/altindag/log/LogCaptor.java [hunk 1]
```diff
@@ -124,20 +135,24 @@ public static LogCaptor forName(String name) {
...
     }
 
     public List<LogEvent> getLogEvents() {
-        return listAppender.list.stream()
-                .map(toLogEvent())
-                .collect(toList());
+        synchronized (eventList) {
+            return eventList.stream()
+                    .map(toLogEvent())
+                    .collect(toList());
```
Comment (Hakky54) [inline src/main/java/nl/altindag/log/LogCaptor.java:150]: Would it be possible for you to refactor this line to `.collect(collectingAndThen(toList(), Collections::unmodifiableList));` It looks like I have overseen this part when I worked on it

File: src/main/java/nl/altindag/log/LogCaptor.java [hunk 2]
```diff
@@ -193,13 +208,13 @@ public void resetLogLevel() {
     }
 
     public void clearLogs() {
-        listAppender.list.clear();
+        eventList.clear();
```
Comment (Hakky54) [inline src/main/java/nl/altindag/log/LogCaptor.java:211]: Should this als be wrapped in a synchronize block?

Comment (AndreyNudko) [reply src/main/java/nl/altindag/log/LogCaptor.java:211]: clear() is synchronized internally

Comment (Hakky54) [reply src/main/java/nl/altindag/log/LogCaptor.java:211]: Ah, yes it is obviously using the syncronized list and therefor it is safe to call. Sorry missed that one

**Comment by Hakky54** [comment]
Thank you @AndreyNudko for fixing this issue! It looks good to me

**Pull request merged**